### ISSUE TITLE
High CPU load <<1fps

### ISSUE THREAD

**Issue by FritsvanCampen** [scalda/FoolCraft#18]
Foolcraft uses 80% CPU. Forge takes minutes to load. Game takes minutes to load. Windows is adding "(not responding)" to the title.

I looked at https://github.com/scalda/FoolCraft/issues/7 but my game is using more than 6GB of memory and it's not even close to being playable.

The first time I ran the game even the main menu was lagging, spiking up to high CPU and not responding. I haven't tried a 3rd time because it just takes soo long to start.

Any ideas?

**Comment by t0xic00** [comment]
I experience the same thing. At first I thought it was a performance problem with my computer, but as soon as I turned my graphic settings down it still did the same thing. It would let me move around a bit before not responding and crashing. I tried not using a biomes o' plenty map but it still resulted in moving for 2 seconds then "Not responding", and crashing.

Any Ideas?

**Comment by scalda** [comment]
Are you using a dedicated GPU or the built in one?

**Comment by RichardMineall** [comment]
I have a Nvidia Geforce GTX 650, a 3.2 Ghz intel i7 8-core CPU, 12GB of RAM, and this modpack manages to use All my CPU and RAM. I can't even load a world. All the mods (Modpack) works fine, and all I can see that it doesn't have that yours does is Veinminer.

**Comment by Oktoberman** [comment]
I'm having a similar problem, Intel I7 860 at 2.8Ghz, GTX 1060, 16gb RAM. The modpack takes up 96-100% of my CPU, as well as 6-8gb of RAM. The title screen lags, the world takes about 5-10 minutes to load, then I get less than 1 frame per second in game.

**Comment by FritsvanCampen** [comment]
I have a GTX1070 (and a AMD-R9/7900 for additional monitors). Do you think it might be trying to render via the CPU? Is there any way I can check that?

I played SkyBlock a few weeks ago without any issues.

**Comment by Oktoberman** [comment]
I tried doubling the allocated memory in the curse launcher to 8gb. The game is now playable, but still studders. CPU usage has dropped from high 90s to 85% at the most.

**Comment by FritsvanCampen** [comment]
I updated my drivers. No change.

**Comment by scalda** [comment]
can any of you please pastebin a log for once you have loaded the pack please

**Comment by saekalive** [comment]
Running Razer Blade 2014 14" ( i7-4702-HQ, 8gb ram, 870m) running off curse launcher. Pack takes forever to edit any video settings, and ive had to close it several times, but the settings change after i load it back up. currently loading a world i just tried to generate for 20 minutes before closing. it did generate the world but upon reloading the pack i get a brown screen displaying 0% trying to load into the world. Currently it's soaking up 82% of my cpu and 4.4 gigs of ram. going to close it and get the logs*

*SO FAR the pack is refusing to use the minecraft game output thing built into the new launcher that curse is using. so far nothing has changed still sitting with the 0% on the brown screen. all for now

**Comment by saekalive** [comment]
another interesting tidbit is curse is running the minecraft launcher as a 32 bit application (identified by task manager). may be relevant.

**Comment by scalda** [comment]
@saekalive can you make sure you are using you dedicated gpu when running the pack and not the inbuilt cpu gpu please?

**Comment by saekalive** [comment]
@scalda i have the minecraft launcher and the java instance set to run using the dedicated graphics via Nvidia control panel. that's the only way i could see it being able to be configured to use it.

**Comment by patcat127** [comment]
For me, setting curse to use the Jar launcher instead of the Native launcher fixed it.  Just thought I could help!

**Comment by saekalive** [comment]
switching to the Jar launcher seems to have solved all the launcher issues but tried loading the world and it still sits there. heres the pastebin with launcher and game output logs @scalda 

http://pastebin.com/FtFPYPF3

**Comment by FritsvanCampen** [comment]
tl;dr. I fixed it for me. I found a memory setting in Curse that I've now set to max (12GB).

I don't know how to get to the "log for once you have loaded the pack".
I don't know how to check that I'm using the dedicated GPU. Curse has no setting for it. Minecraft has no setting for it. I have an NVidia tool that says it's running javaw.exe so I guess it works?

As I was checking the Curse settings I noticed there is a Java memory slider. I didn't expect to find it there. It was set to System default. I set it to max (12GB). That seems to make the game playable. It's using anywhere between 4 and 9GB. The 'recommended' setting is 2-6GB but that doesn't make the game playable at all. Maybe mod packs should request a minimum memory limit?

I tried getting the game to run on 64 bit. The native launcher insists on using 32 bit. I tried to figure out how to get curse to run in 64 bit but I can't find anything. I tried using a custom launcher and that launches the game but it crashes when it finishes loading.

**Comment by saekalive** [comment]
upon investigating i got a world to load up and running the f3 screen. very weird behavior for the ram. seems weird to me that it does this and ive never seen a pack do this with using so much memory and it also rams the cpu up to 99 whenever it hits 88% ram usage as seen in the video below. maybe a certian mod causing issues? 

https://youtu.be/1o9rzljZPfw

**Comment by scalda** [comment]
@saekalive you are using only 4Gb ram which you can clearly see in your video in your F3 screen please raise this to at least 6GB

**Comment by KumaPoi** [comment]
i found the possible solution for dual-type gpu problem specific for Nvidia.
In nvidia control panel, I set (javaw.exe), (curse.exe), (curseui.exe) and (minecraft.exe) to force-use nvidia gpu.
and in curse launcher, hover the cursor to the modpack and click (view profile), click the (......) and choose (profile option).
Uncheck the (use system setting) and select 6GB ram.
This also possible to solve others modpacks that have same problem.
